addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -223,10 +225,14 @@ public HomematicGateway getGateway() {
     /**
      * Updates the thing for the given Homematic device.
      */
-    public void updateThing(HmDevice device) {
+    private void updateThing(HmDevice device) {
         Thing hmThing = getThingByUID(UidUtils.generateThingUID(device, getThing()));
         if (hmThing != null && hmThing.getHandler() != null) {
-            hmThing.getHandler().thingUpdated(hmThing);
+            ThingHandler thingHandler = hmThing.getHandler();
+            thingHandler.thingUpdated(hmThing);
+            for (Channel channel : hmThing.getChannels()) {
+                thingHandler.channelLinked(channel.getUID());
```

Comment:
The usual way though would be externalizing the logic that updates the channel with the values from your domain objects into a private method and call this then - from here and from handleCommand(REFRESH)